src/plugins/intel_cpu/src/emitters/snippets/aarch64/jit_container_emitter.hpp

```diff
@@ -0,0 +1,36 @@
+// Copyright (C) 2023 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+//
+
+#pragma once
+
+#include "emitters/plugin/aarch64/jit_emitter.hpp"
+
+#include "snippets/lowered/linear_ir.hpp"
+
+namespace ov {
+namespace intel_cpu {
+namespace aarch64 {
+
+///
+/// \brief jit_container_emitter designed to wrap Emitters that contain other Emitters (for example, jit_kernel_emitter)
+///  This is needed to provide common interface for register mapping
+/// (abstract to physical) and nested code access.
+///
+class jit_container_emitter: public jit_emitter {
+public:
+    jit_container_emitter(dnnl::impl::cpu::aarch64::jit_generator* h, dnnl::impl::cpu::aarch64::cpu_isa_t isa);
+
+    // mapping info contains abstract_to_physical map + regs_pool
+    using mapping_info = std::pair<std::map<size_t, size_t>, std::vector<size_t>&>;
+
+protected:
+    // maps gpr and vec abstract registers to physical ones.
+    void map_abstract_registers(mapping_info& gpr_map_pool, mapping_info& vec_map_pool, snippets::lowered::LinearIR::container& expressions) const;
+
+    snippets::lowered::LinearIR body;
```

Comment:
@a-sidorova Thanks Alexandra for the comment! I might have missed something. Do you mean remove the member variable "body", and use kernel->region locally when it is needed in several member functions of inherited classes jit_kernel_emitter and jit_kernel_static_emitter. But it seems we still have to save kernel->region to a member variable, since we can not access to kernel->region directly in jit_kernel_emitter::emit_impl. Maybe a little elaboration would help. Thanks in advance!

Reply:
Now I see that you copied its from x64 code 😄 

When I was reviewing the PR, I was confused a little bit because of `jit_container_emitter` has class field `body` that not used in  `jit_container_emitter` interface. It looked like useless class field. But now I see that it's used in derived class `jit_container_emitter::jit_kernel_emitter` and nowhere else. Maybe it makes sense to move this class field `body` from `jit_container_emitter` to `jit_kernel_emitter`? What do you think?

Reply:
Agree and will apply. Thanks a lot Alexandra! :rocket:

Reply:
Applied. Thanks Alexandra!